docs(recipes): make uniform structure part 1 (#3853)
kubeshop/tracetest: docs/docs/examples-tutorials/recipes/running-python-app-with-opentelemetry-collector-and-tracetest.mdx

## Run This Quckstart Example

The example below is provided as part of the Tracetest project. You can download and run the example by following these steps:

Clone the Tracetest project and go to the Node.js Quickstart with Manual Instrumentation:

```bash
git clone https://github.com/kubeshop/tracetest
cd tracetest/examples/quick-start-python
```

Follow these instructions to run the included demo app and TypeScript example:

1. Copy the `.env.template` file to `.env`.
2. Log into the [Tracetest app](https://app.tracetest.io/).
3. This example is configured to use the OpenTelemetry Tracing Backend. Ensure the environment you will be utilizing to run this example is also configured to use the OpenTelemetry Tracing Backend by clicking on Settings, Tracing Backend, OpenTelemetry, Save.
4. Fill out the [token](https://docs.tracetest.io/concepts/environment-tokens) and [API key](https://docs.tracetest.io/concepts/agent) details by editing your `.env` file. You can find these values in the Settings area for your environment.
5. Run `docker compose up -d`.
6. Run tests from the Tracetest Web UI by accessing the app with the URL `http://app:8080/`.

> Follow along with the sections below for an in detail breakdown of what the example you just ran did and how it works.

Reviewer: ```suggestion
Follow along with the sections below for a detailed breakdown of what the example you just ran did and how it works.
```

## Project Structure

The quick start Python project is built with Docker Compose and contains the Tracetest Agent and a Python Flask app.

The `docker-compose.yaml` file in the root directory of the quick start runs the Python app and the [Tracetest Agent](/concepts/agent) setup.

## Configuring the Python App

The Python app is a simple Flask app, contained in the `app.py` file.

Configure the `.env` like shown below.

```bash
OTEL_EXPORTER_OTLP_TRACES_ENDPOINT="http://tracetest-agent:4317/"
TRACETEST_API_KEY="<YOUR_TRACETEST_API_KEY>"
```

> The OpenTelemetry tracing is contained both as automatic and manual instrumentation. Traces will be sent to Tracetest Agent. Here's the `requiements.txt` with OpenTelemetry libraries.

Reviewer: ```suggestion
The OpenTelemetry tracing is contained both as automatic and manual instrumentation. Traces will be sent to the Tracetest Agent. Here's the `requirements.txt` with OpenTelemetry libraries.
```

```text title="requirements.txt"
flask
opentelemetry-distro
opentelemetry-exporter-otlp
```

Automatic instrumentation is configured in the `Dockerfile` and the `docker-compose.yaml`. The `command` field will include `opentelemetry-instrument`.

```Dockerfile title="Dockerfile"
FROM python:3.10.1-slim
WORKDIR /opt/app
COPY . .
RUN pip install --no-cache-dir -r requirements.txt
RUN opentelemetry-bootstrap -a install
EXPOSE 8080
```

> The `docker-compose.yaml` contains just one service for the Python app. The service is stared with the `command` parameter.

Reviewer: ```suggestion
The `docker-compose.yaml` contains just one service for the Python app. The service is started with the `command` parameter.
```

```yaml title="docker-compose.yaml"
command: opentelemetry-instrument --traces_exporter otlp --service_name app --exporter_otlp_endpoint ${OTEL_EXPORTER_OTLP_TRACES_ENDPOINT} --exporter_otlp_insecure true flask run --host=0.0.0.0 --port=8080
```

Manual instrumentation is configured in the `app.py`.

```python
from flask import Flask, request
import json

from opentelemetry import trace
from opentelemetry.sdk.resources import Resource
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import BatchSpanProcessor
from opentelemetry.sdk.trace.export import ConsoleSpanExporter

provider = TracerProvider()
processor = BatchSpanProcessor(ConsoleSpanExporter())
provider.add_span_processor(processor)
trace.set_tracer_provider(provider)
tracer = trace.get_tracer(__name__)
```

## Running the Python App

To start the full setup, run the following command:

```bash
docker compose up -d
```

> There are 3 endpoints in the Flask app. For seeing manual instrumentation trigger the `"/manual"` endpoint. For seeing the automatic instrumentation trigger the `"/automatic"` endpoint respectively. 

Reviewer: ```suggestion
There are 3 endpoints in the Flask app. To see manual instrumentation, trigger the `"/manual"` endpoint. To see the automatic instrumentation, trigger the `"/automatic"` endpoint respectively. 
```
